AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSTypedProperty.h

```diff
@@ -0,0 +1,53 @@
+#import <Foundation/Foundation.h>
+#import "MSSerializableObject.h"
+
+
+@interface MSTypedProperty : NSObject <MSSerializableObject>
+
+/**
+ * Property type.
+ */
+@property(nonatomic, copy) NSString *type;
+
+/**
+* Property name.
+*/
+@property(nonatomic, copy) NSString *name;
+
+/**
+* Property value.
+*/
+@property(nonatomic, copy) NSObject *value;
```

Comment:
Do you have any reasons for the recent change?
This would be dangerous since type and object type could mismatch.
If you don't have any strong reason to do this one, I would like to suggest you to make the class back to abstract and having separate child classes for different types.

Reply:
How can a mismatch occur? I just found that the code was far simpler this way, didn't see the need to introduce so many types when one will suffice. You can also look at how we implemented `MSCustomProperties` - there isn't a separate type for each type